### ISSUE TITLE
datasets.MNIST hangs

### ISSUE THREAD

**Issue by demongolem-biz** [pytorch/vision#7016]
### 🐛 Describe the bug

Pretty simple really.  The code below tries to get code from yann.lecun.com.  From going there myself, 3 of the 4 gzip files are downloadable but one doesn't download correctly or might be corrupt.  This file is train-labels-idx1-ubyte.gz.

I have identified an alternative site to get a good version of this file.  However if I run the below code, the code hangs when trying to fetch the above code.  Hanging should not be the correct operation.  Preferred to visit another location automatically through code or to give up on downloading the file altogether and continue.

I suppose download=False might get me past this. but it does not seem to be too satisfactory of a solution.

```
train_loader = torch.utils.data.DataLoader(
    datasets.MNIST(
        "../data", train=True, download=True, transform=transforms.Compose(
            [transforms.ToTensor(), transforms.Normalize((0.1307,), (0.3081,))]),), 
    batch_size=args.batch_size,
    shuffle=True,
    **kwargs)
```

### Versions

torch is 1.13.0, torchvision is 0.14.0

cc @datumbox @vfdev-5 @pmeier @VitalyFedyunin @ejguan @NivekT

**Comment by pmeier** [comment]
> However if I run the below code, the code hangs when trying to fetch the above code.

There is nothing inside `torchvision.datasets.MNIST` that should cause the process to hang. Let's make sure that the dataset is properly set up and if the error persists, I'll have to transfer back to the `torchdata` team @NivekT.

For that we only need

```py
from torchvision import datasets

root = "../data"
dataset = datasets.MNIST(root, train=True, download=True)
```

> The code below tries to get code from yann.lecun.com. From going there myself, 3 of the 4 gzip files are downloadable but one doesn't download correctly or might be corrupt.

Unable to reproduce this. Download works as expected on my side.

> I have identified an alternative site to get a good version of this file.

Where did you put it? If you can't or don't want to use our download functionality, the files should be placed like

```
$ROOT
└── MNIST
    └── raw
        ├── t10k-images-idx3-ubyte.gz
        ├── t10k-labels-idx1-ubyte.gz
        ├── train-images-idx3-ubyte.gz
        └── train-labels-idx1-ubyte.gz
```

where `$ROOT` is whatever you pass as `root` into the dataset:

https://github.com/pytorch/vision/blob/f93eb8ff5222b4f95d0f030e0f7813de0455ae4e/torchvision/datasets/mnist.py#L23-L24

>  Hanging should not be the correct operation. Preferred to visit another location automatically through code

Agreed. We actually have a mirror in place:

https://github.com/pytorch/vision/blob/f93eb8ff5222b4f95d0f030e0f7813de0455ae4e/torchvision/datasets/mnist.py#L36-L39

> or to give up on downloading the file altogether and continue.

What do you mean by continue? The only option here is to fail loudly, which we do:

https://github.com/pytorch/vision/blob/f93eb8ff5222b4f95d0f030e0f7813de0455ae4e/torchvision/datasets/mnist.py#L101-L102

Otherwise you would silently miss part of the data.

> I suppose download=False might get me past this.

`download=True` does not only download but also decompress the file, i.e. `train-labels-idx1-ubyte.gz` to `train-labels-idx1-ubyte`. If you want to set `download=False`, make sure you replicate this.

**Comment by pmeier** [comment]
Closing due to inactivity. @demongolem-biz if the issue persists, feel free to re-open.
